container-storage-setup.sh

```diff
@@ -731,40 +729,47 @@ create_partition_parted(){
 }
 
 create_partition() {
-  local dev="$1"
+  local dev="$1" part
 
   if [ -x "/usr/sbin/parted" ]; then
     create_partition_parted $dev
   else
     create_partition_sfdisk $dev
   fi
 
+  part=$(query_first_child $dev)
```

Comment:
Also, if you think `udev settle` is needed, then you should call `query_first_child` after calling it, not before.